Content.Server/Connection/ConnectionManager.cs

```diff
@@ -168,20 +174,36 @@ private async Task NetMgrOnConnecting(NetConnectingArgs e)
                 return (ConnectionDenyReason.Ban, message, bans);
             }
 
-            if (_cfg.GetCVar(CCVars.WhitelistEnabled))
+            // Checks for whitelist IF it's enabled AND the user isn't an admin. Admins are always allowed.
+            if (_cfg.GetCVar(CCVars.WhitelistEnabled) && adminData is null)
             {
-                var min = _cfg.GetCVar(CCVars.WhitelistMinPlayers);
-                var max = _cfg.GetCVar(CCVars.WhitelistMaxPlayers);
-                var playerCountValid = _plyMgr.PlayerCount >= min && _plyMgr.PlayerCount < max;
+                var whitelists = _prototypeManager.EnumeratePrototypes<PlayerConnectionWhitelistPrototype>();
+                var whitelistStringList = _cfg.GetCVar(CCVars.WhitelistPrototype);
 
-                if (playerCountValid && await _db.GetWhitelistStatusAsync(userId) == false
-                                     && adminData is null)
+                // The whitelistStringList is a comma separated list of whitelists to check. I loop through every prototype in the list and check if the user is whitelisted.
+                foreach (var whitelistString in whitelistStringList.Split(','))
                 {
-                    var msg = Loc.GetString(_cfg.GetCVar(CCVars.WhitelistReason));
-                    // was the whitelist playercount changed?
-                    if (min > 0 || max < int.MaxValue)
-                        msg += "\n" + Loc.GetString("whitelist-playercount-invalid", ("min", min), ("max", max));
-                    return (ConnectionDenyReason.Whitelist, msg, null);
+                    var whitelist = whitelists.FirstOrDefault(w => w.ID == whitelistString);
+                    if (whitelist is null)
+                    {
+                        _sawmill.Warning($"Whitelist prototype {whitelistString} does not exist.");
+                        continue;
```

Comment:
You should block connecting entirely due to misconfiguration if this error occurs, as a configuration error may open a security hole in the server.